cluster-autoscaler/cloudprovider/mcm/mcm_cloud_provider.go

```diff
@@ -423,7 +440,51 @@ func (machinedeployment *MachineDeployment) Nodes() ([]cloudprovider.Instance, e
 // Implementation optional.
 // TODO: add proper implementation
 func (machinedeployment *MachineDeployment) GetOptions(defaults config.NodeGroupAutoscalingOptions) (*config.NodeGroupAutoscalingOptions, error) {
-	return nil, cloudprovider.ErrNotImplemented
+	mcdSpec, err := machinedeployment.mcmManager.GetMachineDeploymentSpec(machinedeployment.Name)
+	if err != nil {
+		return nil, err
+	}
+	if mcdSpec == nil {
+		return nil, fmt.Errorf("nil machinedeployment returned for %s", machinedeployment.Name)
+	}
```

Comment:
I think this check is unnecessary. If there is no error, then machine deployment will not be nil.